INSTALL.md

```diff
@@ -81,71 +81,96 @@ We also include a script in the repository `install-rust.sh` that downloads and
 
 ### Ubuntu
 
-#### Ubuntu 20.04
+#### Ubuntu 22.04
 You can install the [test toolchain](#adding-the-test-toolchain) to build and run stellar-core with the latest version of the llvm toolchain.
 
-Alternatively, if you want to just depend on stock Ubuntu, you will have to build with clang *and* have use `libc++` instead of `libstdc++` when compiling.
+Alternatively, if you want to just depend on stock Ubuntu, you will have to build with clang *and* have to use `libc++` instead of `libstdc++` when compiling.
 
-Ubuntu 20.04 has clang-12 available, that you can install with
+Ubuntu 22.04 has clang-20 available but you have add LLVM apt repository dependency,
+```zsh
+sudo wget -O /etc/apt/trusted.gpg.d/apt.llvm.org.asc https://apt.llvm.org/llvm-snapshot.gpg.key
+
+sudo tee /etc/apt/sources.list.d/llvm.list <<EOF
+deb http://apt.llvm.org/jammy/ llvm-toolchain-jammy-20 main
+deb-src http://apt.llvm.org/jammy/ llvm-toolchain-jammy-20 main
+EOF
+```
 
-    # install clang-12 toolchain
-    sudo apt-get install clang-12
+After that you can install clang-20,
+```zsh
+sudo apt-get update
+sudo apt-get -y install clang-20 llvm-20
+```
 
 After installing packages, head to [building with clang and libc++](#building-with-clang-and-libc).
 
 
 #### Adding the test toolchain (optional)
-    # NOTE: newer version of the compilers are not
-    #    provided by stock distributions
-    #    and are provided by the /test toolchain
-    sudo apt-get install software-properties-common
-    sudo add-apt-repository ppa:ubuntu-toolchain-r/test
-    sudo apt-get update
+```zsh
+# NOTE: newer version of the compilers are not
+#    provided by stock distributions
+#    and are provided by the /test toolchain
+sudo apt-get install software-properties-common
+sudo add-apt-repository ppa:ubuntu-toolchain-r/test
+sudo apt-get update
+```
```

Comment:
I don't know if there's much point in mentioning the PPA if we're mentioning apt.llvm.org itself?